src/qt3utils/applications/qt3ple/main.py

```diff
@@ -0,0 +1,311 @@
+                    help='NI DAQ terminal connected to input digital TTL signal')
+parser.add_argument('-c', '--clock-rate', metavar = 'rate (Hz)', default = 100000, type=int,
+                    help='''Specifies the clock rate in Hz. If using an external clock,
+                    you should specifiy the clock rate here so that the correct counts per
+                    second are displayed. If using the internal NI DAQ clock (default behavior),
+                    this value specifies the clock rate to use. Per the NI DAQ manual,
+                    use a suitable clock rate for the device for best performance, which is an integer
+                    multiple downsample of the digital sample clock.''')
+parser.add_argument('-n', '--num-data-samples-per-batch', metavar = 'N', default = 250, type=int,
+                    help='''Number of data points to acquire per DAQ batch request.
+                           Note that only ONE data point is shown in the scope.
+                           After each request to the NI DAQ for data, the mean count
+                           rate from the batch is computed and displayed. Increasing
+                           the "num-data-samples-per-batch" should reduce your noise, but
+                           slow the response of the scope. Increase this value if the
+                           scope appears too noisy.''')
+parser.add_argument('-ct', '--clock-terminal', metavar = 'terminal', default = None, type=str,
+                    help='''Specifies the digital input terminal to the NI DAQ to use for a clock.
+                            If None, which is the default, the internal NI DAQ clock is used.''')
+parser.add_argument('-to', '--rwtimeout', metavar = 'seconds', default = 10, type=int,
+                    help='NI DAQ read/write timeout in seconds.')
+parser.add_argument('-sc', '--signal-counter', metavar = 'ctrN', default = 'ctr2', type=str,
+                    help='NI DAQ interal counter (ctr1, ctr2, ctr3, ctr4)')
+parser.add_argument('--wavelength-write-channel', metavar = 'ch0', default = 'ao0', type=str,
+                    help='Analog output channel used to control the wavelength of the laser')
+parser.add_argument('--wavelength-read-channel', metavar = 'ch0', default = 'ai0', type=str,
+                    help='Analog input channels used to read the instantaneous wavelength')
+parser.add_argument('-lmin', '--wavelength-min-position', metavar = 'voltage', default = -10, type=float,
+                    help='sets min allowed voltage on PLE controller.')
+parser.add_argument('-lmax', '--wavelength-max-position', metavar = 'voltage', default = 10, type=float,
+                    help='sets min allowed voltage on PLE controller.')
+parser.add_argument('-lscale', '--wavelength-scale-nm-per-volt', default = 1, type=float,
+                    help='sets nanometer to volt scale for PLE controller.')
+parser.add_argument('-r', '--randomtest', action = 'store_true',
```

Comment:
The application doesn't work as expected when using the -r flag. `qt3ple -r` runs, but raises an exception when 'start scan' button is pressed. (Hmmm... I guess that was one good reason to subclass from nipiezojenapy BaseControl.. :/)